precompile/warp_messenger.go

```diff
@@ -252,7 +255,27 @@ func (c *WarpMessengerConfig) verifyPredicate(predicateContext *PredicateContext
 // OnAccept optionally returns a function to perform on any log with the precompile address.
 // If enabled, this will be called after the block is accepted to perform post-accept computation.
 func (c *WarpMessengerConfig) OnAccept() OnAcceptFunc {
-	return nil
+	return c.onAccept
+}
+
+func (c *WarpMessengerConfig) onAccept(backend Backend, txHash common.Hash, logIndex int, topics []common.Hash, logData []byte) error {
+	if backend == nil {
+		return ErrMissingPrecompileBackend
+	}
+
+	if len(topics) != 3 || topics[0] != common.HexToHash(SubmitMessageEventID) {
+		return ErrInvalidLogTopics
+	}
```

Comment:
Can we split this up so we get a more specific error if something goes wrong and we hit this error?

We should return a formatted error with the number of topics if it's incorrect and the actual hash value of the topic at index 0 if it does not match the expected one.

Reply:
updated to separate errors